R/preview.R

```diff
@@ -248,15 +258,23 @@ st2article <- function(..., .list = NULL, ntex = 1,  #nocov start
   temp <- mgluet(temp, .envir = env)
 
   if(is.character(caption)) {
-    wrap_with_caption <- function(text, i) {
-      if(is.null(i)) i <- "<no name given>"
-      short <- paste0("pmtables output preview - ", i)
+    wrap_with_caption <- function(text, i, short, caption) {
+      if(has.st_caption(text)) {
+        caption <- cap_main(text)
+        short <- cap_short(text)
+        if(is.null(short)) short <- caption
```

Comment:
Just to check my understanding: a NULL `short` isn't actually reachable given the upstream code, right?  Line 217-221 ensures that short is assigned to something non-NULL.

Reply:
The code isn't very clear; there's lots of different things using the same name. 

When you call `cap_short()` on 254, that will go into the table text, pull off the caption and try to pull off the short title. If there was no short title, that should come back null which is why the check is there. The stuff back on 217 - 221 is setting up captions and short names in case `st_caption()` wasn't used; that's the previous behavior that I tried to preserve.

Reply:
> When you call cap_short() on 254 [...]

Doh, I missed that override.  Thanks.